src/hotspot/share/opto/divnode.cpp

```diff
@@ -1520,12 +1520,14 @@ Node* ModFNode::Ideal(PhaseGVN* phase, bool can_reshape) {
   if (!can_reshape) {
     return nullptr;
   }
+  if (proj_out_or_null(TypeFunc::Control) == nullptr) { // dead node
```

Comment:
Interesting. You have extracted a method for `is_unused`, which you use multiple times. But you did not do it for this check which you also use multiple times. You could also wrap this in some method like `is_dead`.